scripts/zones/Aht_Urhgan_Whitegate/npcs/Waoud.lua

```diff
@@ -158,5 +177,19 @@ function onEventFinish(player,csid,option)
         player:messageSpecial(ID.text.PAY_DIVINATION) -- You pay 1000 gil for the divination.
     elseif csid == 707 then
         npcUtil.completeQuest(player, AHT_URHGAN, tpz.quest.id.ahtUrhgan.BEGINNINGS, {item=17717})
+
+    -- OMENS
+    elseif csid == 710 then
+        player:addQuest(AHT_URHGAN,tpz.quest.id.ahtUrhgan.OMENS)
+        player:setCharVar("BluAf2CS",1)
+    elseif csid == 711 and option == 1 then
+        player:delGil(1000)
```

Comment:
No need to check the player's Gil as the check is part of the params for it.

No changes made here.

Reply:
Packet injection is a thing 😉 

When params are sent to the client in a `startEvent`, those params only tell a client how to change the display of the event. The resulting "option" that's sent by the client to the server is entirely under the client's control, and there's no way to prevent / guarantee a client can _only_ send certain options back to the server after being given specific params by us.

So instead, we need to be sure to _never trust the client_ and always re-verify that "meaningful" options they send (ie: resulting in giving items, or in this case, deleting gil) are actually valid and that we're not being lied to.

Reply:
ah got it, but in this case it's deleting gil so I would assume if someone is trying to cheat then they must be really dumb to cheat to delete their items lol.

If we are trying to avoid cheating then it would be for other stuff like unlocking this job without meeting the criteria.

I can go and change this and make all the onEventFinish to have the same requirements that started that event :cat: let me know.

Reply:
after thinking it, you are right @ibm2431 :cat:  it makes sense to not be fooled by those cheaters that like to mess the game experience.

I only included it inside the OMENS onEventFinish. But did it fully on Lathuya. Maybe it could be a good project to be placed on the projects section :sunglasses: